src/server/frontend_xwayland/xwayland_surface.cpp

```diff
@@ -688,12 +688,18 @@ void mf::XWaylandSurface::attach_wl_surface(WlSurface* wl_surface)
 
     std::shared_ptr<XWaylandClientManager::Session> local_client_session;
     std::shared_ptr<ms::Session> session;
+    bool rejected = false;
     reply_functions.push_back(connection->read_property(
         window, connection->_NET_WM_PID,
```

Comment:
We can actually get this generically, rather than relying on the client to set `_NET_WM_PID`, via the XResources extension. Mutter has [some code](https://gitlab.gnome.org/GNOME/mutter/-/commit/e5347af0005a68ea9804690658abd876b2186ab6#abaf68a20e31f3ff44fa4596691ecc717f319bd5_129_194) for this.